components/player-loader.tsx

```diff
@@ -0,0 +1,69 @@
+import { forwardRef } from 'react';
+import { breakpoints } from '../style-vars';
+import { HTMLVideoElementWithPlyr } from '../types';
+import dynamic from 'next/dynamic';
+
+const PlyrPlayer = dynamic(() => import('./plyr-player'));
+
+type Props = {
+  playbackId: string
+  poster: string
+  currentTime?: number
+  aspectRatio: number;
+  onLoaded: () => void;
+  playerType: string;
+  onError: (error: ErrorEvent) => void;
+};
+
+const PlayerLoader = forwardRef<HTMLVideoElementWithPlyr, Props>(({ playbackId, poster, currentTime, aspectRatio, playerType, onLoaded, onError }, ref) => {
+  const isVertical = aspectRatio < 1;
+
+  return (
+    <>
+      <div className='video-container'>
+        {playerType === "plyr" && <PlyrPlayer forwardedRef={ref} isVertical={isVertical} playbackId={playbackId} poster={poster} currentTime={currentTime} onLoaded={onLoaded} onError={onError} /> }
```

Comment:
we can add more conditionals here for different `playerType`s -- `mux-player` and `videojs` would be great to start with